src/dmd/e2ir.d

```diff
@@ -6062,35 +6063,41 @@ Symbol *toStringSymbol(const(char)* str, size_t len, size_t sz)
             /* The stringTab pools common strings within an object file.
              * Win32 and Win64 use COMDATs to pool common strings across object files.
              */
-            import dmd.root.outbuffer : OutBuffer;
-            import dmd.dmangle;
-
-            scope StringExp se = new StringExp(Loc.initial, str[0 .. len], len, cast(ubyte)sz, 'c');
-
             /* VC++ uses a name mangling scheme, for example, "hello" is mangled to:
              * ??_C@_05CJBACGMB@hello?$AA@
              *        ^ length
              *         ^^^^^^^^ 8 byte checksum
              * But the checksum algorithm is unknown. Just invent our own.
              */
+
+            import dmd.root.outbuffer : OutBuffer;
             OutBuffer buf;
             buf.writestring("__");
-            mangleToBuffer(se, &buf);   // recycle how strings are mangled for templates
-
-            if (buf.length >= 32 + 2)
-            {   // Replace long string with hash of that string
-                import dmd.backend.md5;
-                MD5_CTX mdContext = void;
-                MD5Init(&mdContext);
-                MD5Update(&mdContext, cast(ubyte*)buf.peekChars(), cast(uint)buf.length);
-                MD5Final(&mdContext);
-                buf.setsize(2);
-                foreach (u; mdContext.digest)
+
+            void printHash()
+            {
+                // Replace long string with hash of that string
+                // Use the hash calculated in stringTab and the length
+                buf.writestring("ls");
+                buf.print(len);
+                buf.writeByte('_');
+                buf.printHex(hash, 8);
+            }
+
+            const mangleMinLen = 14; // mangling: "__a14_(14*2 chars)" = 6+14*2 = 34
```

Comment:
The comment list 34 but the code uses 14?

Reply:
A string of length 14 produces 34 (or at least 34) bytes in its mangled form.